backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -695,6 +699,42 @@ public FutureExperimentRunDAO(
                                                     hyperparams.get(builder.getId()))),
                                     executor);
 
+                            final var futureHyperparamsFromConfigBlobs =
+                                InternalFuture.completedInternalFuture(
+                                        config.populateConnectionsBasedOnPrivileges)
+                                    .thenCompose(
+                                        populateConnectionsBasedOnPrivileges -> {
+                                          if (populateConnectionsBasedOnPrivileges) {
+                                            return getAllowedEntitiesByResourceType(
+                                                ModelDBActionEnum.ModelDBServiceActions.READ,
+                                                ModelDBResourceEnum.ModelDBServiceResourceTypes
+                                                    .REPOSITORY);
+                                          } else {
+                                            return InternalFuture.completedInternalFuture(
+                                                new ArrayList<>());
+                                          }
+                                        },
+                                        executor)
+                                    .thenCompose(
+                                        repositoryIds ->
+                                            hyperparametersHandler
+                                                .getExperimentRunHyperparameterConfigBlobMap(
+                                                    new ArrayList<>(ids), repositoryIds),
+                                        executor);
+                            futureBuildersStream =
+                                futureBuildersStream.thenCombine(
+                                    futureHyperparamsFromConfigBlobs,
+                                    (stream, hyperparamsFromConfigBlob) ->
+                                        stream.peek(
```

Comment:
map, don't `peek` and then do inplace operation

Reply:
Implemented with map